src/main.rs

```diff
@@ -282,7 +286,11 @@ async fn search(args: &Args, stopped: Arc<AtomicBool>) -> Result<()> {
 
     let mut search_stream = futures::stream::iter(query_stream).buffer_unordered(args.parallel);
     while let Some(result) = search_stream.next().await {
-        result?;
+        if !args.ignore_errors {
+            result?;
+        } else if let Err(err) = result {
+            progress_bar.println(format!("Error: {}", err));
+        }
```

Comment:
Easier to read this way, IMO:

```suggestion
        if args.ignore_errors {
            if let Err(err) = result {
                progress_bar.println(format!("Error: {}", err));
            }
        } else {
            result?;
        }
```

Or this:

```suggestion
        // Handle error, if any...
        let err = match result {
            Err(err) => err,
            Ok(()) => continue, // `Ok(_)`, if `result` is not `Result<(), _>`
        };
        
        if args.ignore_errors {
            progress_bar.println(format!("Error: {}", err));
        } else {
            return Err(err); // `err.into()`, if error types don't match exactly
        }
```

Reply:
I like the early return (the latter) one better. Thanks!